cpp/util/SnowflakeCommon.cpp

```diff
@@ -108,10 +108,13 @@ CURLcode set_curl_proxy(CURL *curl, const char* proxy, const char* no_proxy)
     if (res != CURLE_OK) return res;
     res = curl_easy_setopt(curl, CURLOPT_PROXYPORT, (long)proxySettings.getPort());
     if (res != CURLE_OK) return res;
-    res = curl_easy_setopt(curl, CURLOPT_PROXYUSERNAME, proxySettings.getUser().c_str());
-    if (res != CURLE_OK) return res;
-    res = curl_easy_setopt(curl, CURLOPT_PROXYPASSWORD, proxySettings.getPwd().c_str());
-    if (res != CURLE_OK) return res;
+    if (!proxySettings.getUser().empty() || !proxySettings.getPwd().empty())
```

Comment:
Do we need both user name and password? if yes, then it should be AND condition.

Reply:
It's possible that have valid user name and empty password. The current logic is that when either username or password is set, Proxy-Authorization header will be added to with the settings, the one not being set will be empty.

Reply:
The fix in aws-sdk-cpp [here](https://github.com/aws/aws-sdk-cpp/commit/59fe5b1b4d3cf9d26586a543376818e581f5aafc) is using OR condition as well so I think using the same logic would be safe.

Reply:
ok. But what will happen user name is empty but the password is provided?